rfcs/min-relations.md

```diff
@@ -0,0 +1,438 @@
+            amount_owed *= 1.05; 
+        }
+    }
+}
+
+// You can query for entities that target a specific entity
+fn friend_of_dorothy(
+    mut query: Query<Entity, With<Relation<FriendsWith>>>, 
+    dorothy: Res<Dorothy>,
+) {
+    // Setting relation filters requires mutable access to the query
+    query.filter_relation::<FriendsWith, _>(
+        RelationFilter::target(dorothy.entity)
+    // .build() causes the relation filters to be applied
+    ).build();
+
+    for source_entity in query.iter(){
+        println!("{} is friends with Dorothy!", source_entity);
+    }
+
+    // RelationFilters are reset at the end of the system, make sure to set them every time your system runs!
+    // You can also reset this manually by applying a new `RelationFilters` to the query
+    // They override existing relation filters, rather than adding to them
+}
+
+// Or even look for some combination of targets!
+fn caught_in_the_middle(mut query: Query<&mut Stress, With<Relation<FriendsWith>>>,
+    dorothy: Res<Dorothy>,
+    elphaba: Res<Elphaba>){
+
+    // Note that we can set relation filters even if the relation is in a query filter
+    query.filter_relation::<FriendsWith, _>(
+        RelationFilter::any_of()
+        .target(dorothy.entity)
+        .target(elphaba.entity)
+    // You can directly chain this as set_relation_filters returns &mut Query
+    ).build().for_each(|mut stress| {
+        println!("{} is friends with Dorothy and Elphaba!", source_entity);
+        stress.val = 1000; 
+    })
+}
+
+// Query filters work too!
+fn not_alone(
+    commands: mut Commands, 
+    query: Query<Entity, Without<Relation<FriendsWith>>>,
+    puppy: Res<Puppy>
+) {
+    for lonely_entity in query.iter(){
+        commands.insert_relation(FriendsWith, puppy.entity);
+    }
+}
+
+// So does change detection with Added and Changed!
+fn new_friends(query: Query<&mut Excitement, Added<Relation<FriendsWith>>>){
+    query.for_each(|(mut excitement, new_friends)| {
+        excitement.value += 10 * new_friends.count();
+    });
+}
+
+```
+
+You can use the `Entity` returned by your relations to fetch data from the target's components by combining it with `query::get()` or `query::get_component<C>()`.
+
+```rust
+fn debts_come_due(
+    mut debt_query: Query<(Entity, Relation<Owes>)>, 
+    mut money_query: Query<&mut Money>,
+) {
+    for (debtor, debt) in debt.query() {
+        for (lender, amount_owed) in debt {
+            *money_query.get(debtor).unwrap()
+                .checked_sub(amount_owed)
+                .expect("Nice kneecaps you got there...");
+            *money_query.get(lender).unwrap() += amount_owed;
+        }
+    }
+}
+```
+
+### Nuances of relations
+
+Relations between entities are automatically cleaned up when either the source or target entity is removed.
+Just like components, you can view which relations were removed in the last frame through the `RemovedRelation<T>` system parameter, which also returns their data.
```

Comment:
Why is `RemovedRelation` a separate thing? Should we use `Removed<Relation<T>>` instead?

Reply:
I would prefer this, but I'm following the convention from components.

Reply:
I see, it's not a query combinator, but a system param. Then the convention really should be `RemovedRelations<T>`, which probably would be an iterator over `(Entity, RemovedTargets)` where `RemovedTargets` is another iterator over `Entity` (or whatever the associated Target type will be in the future).

Reply:
Ah, you're right there. I'll add the `s`. Down the line, I'd prefer to see Removed* move to subworlds, but that's out of scope here.